<launch>
	<!-- Launch the motion detector keyboard for user input. -->
	<!-- Use launch-prefix to launch this node in a separate window. -->
	<node pkg="motion_detector" type="motion_detector_keyboard_node" name="motion_detector_keyboard_node" output="screen" launch-prefix="xterm -e" />

	<!-- Launch the driver node for our usb camera. -->
	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
		<param name="video_device" value="/dev/video0" />
		<param name="image_width" value="320" />
		<param name="image_height" value="240" />
		<!--
		<param name="pixel_format" value="mjpeg" />
		-->
		<param name="pixel_format" value="yuyv" />
		<param name="camera_frame_id" value="usb_cam" />
		<param name="io_method" value="mmap" />
  </node> 
 
	<!-- Launch the motion detector node for image processing. -->
	<node name="motion_detector_python" pkg="motion_detector" type="motion_detector_python.py" />

	<!-- Launch the image view to display the video. -->
	<node pkg="image_view" type="image_view" name="image_view" output="screen">
		<!-- Set the window name. -->
		<param name="window_name" value="Motion_Detector" />
		<!-- Remap from default image topic to the topic provided by usb_cam. -->
		<remap from="/image" to="/image_converter/output_video"/>
	</node>
</launch>
